src/development/coding-logging.md

```diff
@@ -0,0 +1,251 @@
+---
+layout: article
+title: "Logging"
+tags: dev
+order: 5
+---
+Understand the reason for logging within a component and write log statements appropriate to these reasons.
+
+Reasons may include:
+
+* __Application support__
+  Logs can be used to track down issues in production. It should be possible to read logs and understand the flow through a programme to help track down where something may have gone wrong.
+* __Application monitoring__
+  Logs can be monitored and used to trigger alerts when something goes wrong.Logs should be monitored for application errors and security events at a minimum. Client errors may not be actively monitored, although they should be logged.
+* __Application profiling__
+  Logs can be used to understand how an application is behaving in production. For instance to identify bottlenecks or performance issues.
+* __Audit__
+  Care must be taken when considering logging for audit purposes. __Personally Identifiable Information must not be recorded in logs__. Logs should only contain an audit of events, identified by system generated IDs. For instance:
+  `User [id] logged into [application-a]`
```

Comment:
Would we want to mention DataDog here or keep it generic?
If we follow DataDog logging conventions here it would be like `User: [id] logged into application: [application-a]` which would give access to more features in parsing the logs there.

Reply:
I think we should keep this bit generic, but we should also call out our standard tools in this doc.

Reply:
I see we have Datadog linked already in the Tools section.